src/common/QskRgbValue.h

```diff
@@ -235,4 +235,335 @@ namespace QskRgb
+                const auto r = ( argb >> 16 ) & 0xFF;
+                const auto g = ( argb >> 8 ) & 0xFF;
+                const auto b = ( argb >> 0 ) & 0xFF;
+                const auto a = ( len == max ? argb >> 24 : 0xFF ) & 0xFF;
+                return qRgba( r, g, b, a );
+            }
+
+            // converts a hex literal from '0xRRGGBB[AA]' to '0xAARRGGBB' integer
+            template< char... Cs >
+            constexpr QRgb operator""_rgba()
+            {
+                constexpr auto rrggbb = 8;
+                constexpr auto rrggbbaa = 10;
+                static_assert( sizeof...( Cs ) == rrggbb || sizeof...( Cs ) == rrggbbaa,
+                    "invalid color literal length" );
+                constexpr auto rgba = parse_hex_literal_0< Cs... >();
+                if constexpr ( sizeof...( Cs ) == rrggbb )
+                {
+                    const auto r = ( rgba >> ( 4 * 4 ) ) & 0xFF;
+                    const auto g = ( rgba >> ( 2 * 4 ) ) & 0xFF;
+                    const auto b = ( rgba >> ( 0 * 4 ) ) & 0xFF;
+                    const auto a = 0xFF;
+                    return qRgba( r, g, b, a );
+                }
+                if constexpr ( sizeof...( Cs ) == rrggbbaa )
+                {
+                    const auto r = ( rgba >> ( 6 * 4 ) ) & 0xFF;
+                    const auto g = ( rgba >> ( 4 * 4 ) ) & 0xFF;
+                    const auto b = ( rgba >> ( 2 * 4 ) ) & 0xFF;
+                    const auto a = ( rgba >> ( 0 * 4 ) ) & 0xFF;
+                    return qRgba( r, g, b, a );
+                }
+            }
+
+            // converts a hex literal from '0x[AA]RRGGBB' to '0xAARRGGBB' integer
+            template< char... Cs >
+            constexpr QRgb operator""_argb()
+            {
+                constexpr auto rrggbb = 8;
+                constexpr auto aarrggbb = 10;
+                static_assert( sizeof...( Cs ) == rrggbb || sizeof...( Cs ) == aarrggbb,
+                    "invalid color literal length" );
+                constexpr auto rgba = parse_hex_literal_0< Cs... >();
+                if constexpr ( sizeof...( Cs ) == rrggbb )
+                {
+                    const auto r = ( rgba >> ( 4 * 4 ) ) & 0xFF;
+                    const auto g = ( rgba >> ( 2 * 4 ) ) & 0xFF;
+                    const auto b = ( rgba >> ( 0 * 4 ) ) & 0xFF;
+                    const auto a = 0xFF;
+                    return qRgba( r, g, b, a );
+                }
+                if constexpr ( sizeof...( Cs ) == aarrggbb )
+                {
+                    const auto r = ( rgba >> ( 4 * 4 ) ) & 0xFF;
+                    const auto g = ( rgba >> ( 2 * 4 ) ) & 0xFF;
+                    const auto b = ( rgba >> ( 0 * 4 ) ) & 0xFF;
+                    const auto a = ( rgba >> ( 6 * 4 ) ) & 0xFF;
+                    return qRgba( r, g, b, a );
+                }
+            }
+
+#define QSK_REQUIRE_CONSTEXPR_LITERAL 1
+#ifdef QSK_REQUIRE_CONSTEXPR_LITERAL
```

Comment:
@uwerat how do we want to proceed with this? I think it can stay and the user can decide to add `DEFINES += QSK_REQUIRE_CONSTEXPR_LITERAL` in order to guarantee `constexpr` literals.

Obviously the line above must be removed before merging: `#define QSK_REQUIRE_CONSTEXPR_LITERAL 1`

Reply:
Why not moving these assertions to QskRgbValue.cpp. It should be good enough to do them once - when building the qskinny lib.

But in the end I do not believe, that we need to keep these assertions at all. It is not very likely that someone will touch the code after it is working once and in case it is necessary it is usually because of something that has not been found by those checks.

Reply:
The idea was to use the assertion both as constexpr check and documentation / example, but it's fine for me to move it into the cpp. Although when thinking about prebuilt packaged the user would have to write his own checks...

The whole static assertion is just to guarantee that the literals are constexpr since the keyword itself doesn't guarantee it.

Reply:
Then it would be good enough to have only one static assertion. But in case there is is some compiler that fails, what is the user supposed to do with this information ?

IMHO the checks are not worth to introduce a compile time penalty to each file that is including the literals.